Drop few unnecessary CONFIG_PINMUX=y and pinmux.h includes

fabiobaltieri opened the pull request:
Drop few CONFIG_PINMUX=y and pinmux.h includes, one more step towards dropping that code.

fabiobaltieri committed:
board: v2m_musca: drop CONFIG_PINMUX=y

Drop CONFIG_PINMUX=y from v2m_musca_b1 and v2m_musca_s1. There's no
PINMUX driver for these and the board specific pinmux.c is always
enabled, so this should have no effect.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
board: v2m_beetle: drop CONFIG_PINMUX=y

There's no actual pinmux subsystem driver for v2m_beetle, and
CONFIG_PINMUX is only used as a proxy to build pinmux.c.

Drop the config and change the file to be built unconditionally (same as
mps2_an521 and mps2_an385.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

gmarull commented on boards/arm/v2m_beetle/CMakeLists.txt:
```diff
@@ -1,7 +1,5 @@
-  zephyr_library_sources(pinmux.c)
-endif()
+zephyr_library()
+zephyr_library_include_directories(${ZEPHYR_BASE}/drivers)
```
this looks suspicious, can likely be removed

fabiobaltieri replied:
yeah I tried, fails the build on `#include <gpio/gpio_cmsdk_ahb.h>`, somehow it's including the internal header, should be split out in `include/drivers`, won't bother for this PR :-)

fabiobaltieri replied:
Well I added a patch for that.

gmarull commented on boards/arm/v2m_beetle/CMakeLists.txt:
```diff
@@ -1,7 +1,5 @@
-endif()
+zephyr_library()
+zephyr_library_include_directories(${ZEPHYR_BASE}/drivers)
+zephyr_library_sources(pinmux.c)
```
I see this board is including drivers/pinmux.h for nothing, can be removed as well.

fabiobaltieri replied:
you are right, good point, all of them in fact, and other boards too... I'll have a wider look and clean those up as well

gmarull commented on boards/arm/v2m_beetle/Kconfig.defconfig:
```diff
@@ -8,10 +8,6 @@ if BOARD_V2M_BEETLE
 config BOARD
 	default "v2m_beetle"
 
-config PINMUX_BEETLE
```
wondering if the pinmux driver should be deleted then...

fabiobaltieri replied:
oh, there's a Kconfig.beetle... yeah I think that should go as well

fabiobaltieri committed:
boards: drop pinmux subsystem references from board code

Drop few references to pinmux.h from board pinmux specific code. The
header is not neeeded since no pinmux subsystem functions are used here.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
drivers: drop few unused pinmux.h include

Drop few unused include pinmux header from drivers that don't seem to be
using it anymore.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri commented:
CI fail needs a trusted-firmware-m patch...

erwango commented on drivers/gpio/gpio_stm32.c:
```diff
@@ -18,8 +18,6 @@
 #include <stm32_ll_system.h>
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/drivers/clock_control/stm32_clock_control.h>
-#include <pinmux/pinmux_stm32.h>
```
We're actually using it but you can remove this one since it's also included in gpio_stm32.h.
But yes, once PINMUX deprecation period is complete, we'll need to move definitions from pinmux_stm32.h  to gpio_stm32.h

erwango reviewed (dismissed):
Ok for STM32

fabiobaltieri committed:
gpio: move gpio_cmsdk_ahb.h from drivers to include/drivers

Move gpio_cmsdk_ahb.h out of drivers/gpio and into include/drivers/gpio.
This is used by few board files so it should be in include.

This allows to remove few odd
"zephyr_library_include_directories(${ZEPHYR_BASE}/drivers)".

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri commented:
Update:
- updated west.yaml for the tfm build issue
- moved a gpio header around to drop those odd include headers in the board files

This is still not ready though, I still expect it to fail on some NXP platform, I think that the recent Kconfig+dt changes enabled some extra test combination for stuff like flash + secondary cpus that does not build, may need some dts bit. Will wait for a pass and then sor those out.

fabiobaltieri commented:
Oh I see what's going on, we have two drivers depending on DT_HAS_NXP_LPC_IAP_ENABLED since https://github.com/zephyrproject-rtos/zephyr/pull/48154

```
config SOC_FLASH_LPC
        bool "LPC flash shim driver"
        default y
        depends on DT_HAS_NXP_LPC_IAP_ENABLED
```

and

```
config SOC_FLASH_MCUX
        bool "MCUX flash shim driver"
        default y
        depends on DT_HAS_NXP_LPC_IAP_ENABLED || \
...
```

Guess there should be a platform filter there.

microbuilder commented:
@fabiobaltieri I've included the TF-M manifest update here: https://github.com/zephyrproject-rtos/zephyr/pull/49176

fabiobaltieri commented:
> @fabiobaltieri I've included the TF-M manifest update here: #49176

Thanks, this keeps hitting more unrelated breakages. :-(

gmarull commented:
> > @fabiobaltieri I've included the TF-M manifest update here: #49176
> 
> Thanks, this keeps hitting more unrelated breakages. :-(

pinmux is resisting to die ;-)

fabiobaltieri commented:
@erwango can you re-approve? Same patches just rebased to address unrelated platform breakages.

fabiobaltieri commented:
Okaythanks :-)
